sdk/compiler/damlc/daml-ide/src/DA/Daml/LanguageServer/SplitGotoDefinition.hs

```diff
@@ -216,16 +219,15 @@ gotoDefinitionByName ideState params = do
     file <-
       hoistMaybe (Just "Failed to find module") $
         find (isSuffixOf moduleSuffix . ("/" <> ) . fromNormalizedFilePath) sortedSrcFiles
+    tc <- useOrThrow "Failed to typecheck" TypeCheck file
```

Comment:
Looking through the implementation of `GetSpanInfo`:
```
        tc <- use_ TypeCheck file
        deps <- maybe (TransitiveDependencies [] [] []) fst <$> useWithStale GetDependencies file
        parsedDeps <- mapMaybe (fmap fst) <$> usesWithStale GetParsedModule (transitiveModuleDeps deps)
        (fileImports, _) <- use_ GetLocatedImports file
        packageState <- hscEnv <$> use_ GhcSession file
        let imports = second (fmap artifactFilePath) <$> fileImports
        x <- liftIO $ getSrcSpanInfos packageState imports tc parsedDeps
        return ([], Just x)
```
`getSrcSpanInfos` also extracts definitions from the dependencies of the module, whereas this PR doesn't, is this a regression or do we expect that goto def could'n't've ever worked with deps because of how DARs work?

Note to future self: replacing `GetSpanInfo`  with `TypeCheck` here is fine because GetSpanInfo rule calls Typecheck instantly.

Reply:
I do not want to consider dependencies for the module, goto definition should give the source module, and the correct module will already have been resolved. GHC will already have given the correct module if in the same package, and dars do not look at re-exports, they just give the source, so external modules are also already handled.

Reply:
I will still check though to ensure this is the case.

Reply:
I've verified the logic works as expected